### PR TITLE
MLZFD-14009: Training - Completed Assignment 1-P3/4 [1.0]

### PR DESCRIPTION
**Pull request by Jtauzin**
Added retrofit library to project. Replaced raw json handling with retrofit.

### CODE REVIEW

File: app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/DataResult.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		this.collection = collection;
+	}
+
+
+
+	public ArrayList<NasaImage> getNasaImages () {
+		ArrayList<NasaImage> nasaImages = new ArrayList<NasaImage>();
+		for (Item item : this.collection.getItems()) {
+			try {
+				NasaImage nasaImage = new NasaImage(
```
Comment (pablo-campos) [inline app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/DataResult.java]: This part right here to me seems a little sloppy. You have all data, Why don't you just use the classes you have? The names of the classes you can readjust. The serialization on data fields is just based on the Retrofit annotations, if that makes sense.

File: app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/MainActivity.kt [hunk 2]
```diff
@@ -121,65 +104,34 @@ class MainActivity : AppCompatActivity(), LoaderManager.LoaderCallbacks<ArrayLis
 		connectivityManager.registerDefaultNetworkCallback(object : ConnectivityManager.NetworkCallback() {
 			override fun onAvailable(network: Network) {
 				runOnUiThread(Runnable {
-					if (!isLoaded) {
+					if (imageList.size == 0) {
```
Comment (pablo-campos) [inline app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/MainActivity.kt]: This is NOT good. It's an array list. Learn to use this class.

File: app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/MainActivity.kt [hunk 3]
```diff
@@ -121,65 +104,34 @@ class MainActivity : AppCompatActivity(), LoaderManager.LoaderCallbacks<ArrayLis
...
-					if (imageRecyclerView.adapter == null) {
-						errorView.visibility = View.VISIBLE
-						imageRecyclerView.visibility = View.INVISIBLE
+					if (imageList.size == 0) {
+						adjustVisibility(ERROR)
 					}
 				})
 			}
 		})
+		if (imageList.size == 0) {
```
Comment (pablo-campos) [inline app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/MainActivity.kt]: Same here.

File: app/src/main/java/com/bronzeswordstudios/nasaimageviewer/network/API.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import com.bronzeswordstudios.nasaimageviewer.model.DataResult;
+
+import retrofit2.Call;
+import retrofit2.http.GET;
+import retrofit2.http.Query;
+
+public interface API {
+
+	@GET("search?q=galaxy&media_type=image")
+	Call<DataResult> getVariety0 ();
```
Comment (pablo-campos) [inline app/src/main/java/com/bronzeswordstudios/nasaimageviewer/network/API.java]: Is there not a way to pass the search as a parameter and you do this on your end? 15 methods that are doing the same search you do below is just an overkill. You can literally use the method you have at the bottom: getSearch()

Comment (Jtauzin) [reply app/src/main/java/com/bronzeswordstudios/nasaimageviewer/network/API.java]: Addressed in commit 9322ec1!

File: app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/MainActivity.kt (hunk 3 above)
Comment (Jtauzin) [reply app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/MainActivity.kt]: Addressed in commit 9322ec1!

File: app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/MainActivity.kt (hunk 2 above)
Comment (Jtauzin) [reply app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/MainActivity.kt]: Addressed in commit 9322ec1!

File: app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/DataResult.java (hunk 1 above)
Comment (Jtauzin) [reply app/src/main/java/com/bronzeswordstudios/nasaimageviewer/model/DataResult.java]: Addressed in commit 4589813!

**Pull request merged**